### PR TITLE
Applies rotation to joint to fit the joint axis in URDF

### PR DESCRIPTION
**Pull request by michalpelka**
Signed-off-by: Michał Pełka <michalpelka@gmail.com>

### CODE REVIEW

**Comment by michalpelka** [comment]
Fix to #167

File: Code/Source/RobotImporter/URDF/JointsMaker.cpp [hunk 1]
```diff
@@ -74,7 +74,23 @@ namespace ROS2
         PhysX::EditorJointComponent* jointComponent = nullptr;
         // TODO - ATM, there is no support withing Joint Components for the following:
         // TODO <calibration> <dynamics> <mimic>, friction, effort, velocity, joint safety and several joint types
-        // TODO - apply <axis>
+
+        // WISDOM BELOW:
+        // URDF has rotation axis around Z+ o3de has rotation axis aroud X+
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/JointsMaker.cpp]: All comments: fix punctuation, grammar

File: Code/Source/RobotImporter/URDF/JointsMaker.cpp [hunk 2]
```diff
@@ -74,7 +74,23 @@ namespace ROS2
...
+        // we need to find quaternion that transform X+ to Z+(or given axis) and then convert it to Euler (stinks here!)
+        // TODO do not like conversion to Euler - we should switch to quaternion here!
+        const AZ::Vector3 o3de_joint_dir{ 1.0, 0, 0.0 };
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/JointsMaker.cpp]: middle zero without fractional part

File: Code/Source/RobotImporter/URDF/JointsMaker.cpp [hunk 3]
```diff
@@ -74,7 +74,23 @@ namespace ROS2
         PhysX::EditorJointComponent* jointComponent = nullptr;
         // TODO - ATM, there is no support withing Joint Components for the following:
         // TODO <calibration> <dynamics> <mimic>, friction, effort, velocity, joint safety and several joint types
-        // TODO - apply <axis>
+
+        // WISDOM BELOW:
+        // URDF has rotation axis around Z+ o3de has rotation axis aroud X+
+        // we need to find quaternion that transform X+ to Z+(or given axis) and then convert it to Euler (stinks here!)
+        // TODO do not like conversion to Euler - we should switch to quaternion here!
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/JointsMaker.cpp]: rephrase it (more professional). TODO can comment on how to improve and why.

File: Code/Source/RobotImporter/URDF/JointsMaker.cpp [hunk 4]
```diff
@@ -74,7 +74,23 @@ namespace ROS2
...
+        const AZ::Vector3 o3de_joint_dir{ 1.0, 0, 0.0 };
+        const AZ::Vector3 joint_axis = ROS2::URDF::TypeConversions::ConvertVector3(joint->axis);
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/JointsMaker.cpp]: ROS2 part of the namespace is superfluous

File: Code/Source/RobotImporter/URDF/JointsMaker.cpp (hunk 3 above)
Comment (adamdbrw) [reply Code/Source/RobotImporter/URDF/JointsMaker.cpp]: Also, include the "why" of what we are doing here - in user language.

File: Code/Source/RobotImporter/URDF/JointsMaker.cpp [hunk 5]
```diff
@@ -86,23 +102,48 @@ namespace ROS2
...
+
                 followColliderEntity->Deactivate();
             }
             break;
         case urdf::Joint::REVOLUTE:
             { // Hinge
                 jointComponent = followColliderEntity->CreateComponent<PhysX::EditorHingeJointComponent>();
                 followColliderEntity->Activate();
+
+                const double limit_upper = AZ::RadToDeg(joint->limits->upper);
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/JointsMaker.cpp]: use naming standard here: limitUpper (you can also name it limitUpperDegrees for full clarity)

**Comment by michalpelka** [comment]
Thank you @adamdbrw for your review. I agree with all you comments.
Please find my fixes.

**Review by adamdbrw — APPROVED** [review]

**Pull request merged**